osu.Game/Online/Chat/DrawableLinkCompiler.cs

```diff
@@ -31,10 +32,16 @@ public DrawableLinkCompiler(IEnumerable<Drawable> parts)
             Parts = parts.ToList();
         }
 
+        [Resolved(CanBeNull = true)]
+        private OverlayColourProvider colourProvider { get; set; }
+
+        [Resolved]
+        private OsuColour colours { get; set; }
```

Comment:
Why these have been moved outside of `load()`?

Reply:
Also mapper name colour in Most played beatmaps in UserOverlay is a bit off

Reply:
ColourProvider had to be moved out because it can be null (unless you can do that directly in `load()`?) and OsuColour got moved out for consistency, but you're right, it could've stayed where it was

Reply:
you can make stuff nullable in load (check `AppendComments` in `CommentsPage.cs`)

Reply:
thanks, I should've checked that earlier. i'll fix that and will look for any other colours being off due to this PR.

Reply:
FWIW there's been a PR merged recently that replaced BDL injection with `[Resolved]` wherever possible (#7837). I would like to avoid going back and forth on this stylistic choice.

Reply:
yup, they aren't needed anywhere outside the `load()` so i'll move them back there.